cachito/web/api_v1.py

```diff
@@ -35,12 +40,41 @@ def get_request(request_id):
     :raise NotFound: if the request is not found
     :raise ValidationError: if the request ID is of invalid format
     """
-    try:
-        request_id = int(request_id)
-    except ValueError:
-        raise ValidationError(f'{request_id} is not a valid request ID')
-    request = Request.query.get_or_404(request_id)
-    return flask.jsonify(request.to_json())
+    return flask.jsonify(_get_request_obj(request_id).to_json())
+
+
+@api_v1.route('/requests/<request_id>/download', methods=['GET'])
+def download_archive(request_id):
+    """
+    Download archive of source code.
+
+    :param int request_id: the value of the request ID
+    :return: a Flask send_file response
+    :rtype: flask.Response
+    :raise NotFound: if the request is not found
+    :raise ValidationError: if the request ID is of invalid format
+    """
+    request = _get_request_obj(request_id)
+    # TODO: Verify request has already been processed.
+
+    app_archive_path = Git(request.repo, request.ref).archive_path
+
+    cachito_shared_dir = get_worker_config().cachito_shared_dir
```

Comment:
This makes configuring the API split between the Flask configuration and the Celery configuration. Ideally, the Celery configuration on the API should simply contain the connection details to send requests to the workers. Can this just be a Flask configuration called the same thing?

Reply:
Removed dependency.